### PR TITLE
DOP-4198: New repo docs-atlas-operator

### PR DESCRIPTION
**Pull request by mmeigs**
Creates a `Makefile` and publishedbranches yaml for new repo `docs-atlas-operator`.

This was based off of the `Makefile` for `docs-atlas-cli` because it seemed the most similar. 

This is my first time doing this. Let me know if I'm way off the mark.

### CODE REVIEW

File: publishedbranches/docs-atlas-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+prefix: 'atlas-operator'
```
Comment (jwilliams-mongo) [inline publishedbranches/docs-atlas-operator.yaml:1]: question: when we add a new branch in Hapley will this be updated?

Comment (mmeigs) [reply publishedbranches/docs-atlas-operator.yaml:1]: No, that process is not automated actually. Each of these are added manually for now.

Comment (schmalliso) [reply publishedbranches/docs-atlas-operator.yaml:1]: @mmeigs This file isn't used/needed anymore (we source this info from Hapley), so we can skip it entirely.

File: makefiles/Makefile.docs-atlas-operator [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		mut-publish public ${BUCKET} --prefix="${COMMIT_HASH}/${PATCH_ID}/${MUT_PREFIX}" --stage ${ARGS}; \
+		echo "Hosted at ${URL}/${COMMIT_HASH}/${PATCH_ID}/${MUT_PREFIX}/${USER}/${GIT_BRANCH}/"; \
+	# stagel commit jobs and regular git push jobs\
+	else \
+		mut-publish public ${BUCKET} --prefix="${MUT_PREFIX}" --stage ${ARGS}; \
+		echo "Hosted at ${URL}/${MUT_PREFIX}/${USER}/${GIT_BRANCH}/"; \
+	fi
+
+get-build-dependencies:
+  @curl https://raw.githubusercontent.com/mongodb/docs-worker-pool/meta/publishedbranches/docs-atlas-operator.yaml > ${REPO_DIR}/published-branches.yaml
```
Comment (schmalliso) [inline makefiles/Makefile.docs-atlas-operator:72]:
```suggestion
  echo "Published branches information stored in Atlas collection"
```

**Review by schmalliso — CHANGES_REQUESTED** [review]

File: publishedbranches/docs-atlas-operator.yaml (hunk 1 above)
Comment (mmeigs) [reply publishedbranches/docs-atlas-operator.yaml:1]: Nevermind! See below!

**Review by jwilliams-mongo — APPROVED** [review]
@mmeigs lgtm thanks

**Pull request merged**